docs/release-notes/charmcraft-3.4.0.rst

```diff
@@ -0,0 +1,145 @@
+channel with the same resources.
+
+Feature deprecations
+--------------------
+
+The following features are deprecated in Charmcraft 3.4 and will be removed in
+Charmcraft 4.0:
+
+Windows support
+~~~~~~~~~~~~~~~
+
+Charmcraft 3 has deprecated support running on Windows, and the upcoming 4.0
+release will not have any amount of support for it. Please use
+`Ubuntu inside WSL <https://ubuntu.com/desktop/wsl>`_.
+
+Related issue: `#1810 <https://github.com/canonical/charmcraft/issues/1810>`_
+
+CentOS 7 support
+~~~~~~~~~~~~~~~~
+
+CentOS 7 reached its `End of life
+<https://www.redhat.com/en/topics/linux/centos-linux-eol>`_ on 30 June 2024.
+Charmcraft 3 will continue its current support for CentOS 7, but Charmcraft 4
+will not support CentOS 7.
+
+Related issue: `#1826 <https://github.com/canonical/charmcraft/issues/1826>`_
+
+Bundle registration
+~~~~~~~~~~~~~~~~~~~
+
+New bundle registration `ceased on 1 November 2024
+<https://discourse.charmhub.io/t/15344>`_
+and Charmcraft's ``register-bundle`` command currently returns an error.
+It will be removed entirely in Charmcraft 4.0
+
+Related issue: `#1858 <https://github.com/canonical/charmcraft/issues/1858>`_
+
+Scheduled feature deprecations
+------------------------------
+
+The following features will be deprecated in Charmcraft 4.0:
+
+
+Bundle management
+~~~~~~~~~~~~~~~~~
+
+In line with the phasing out of bundles, Charmcraft 4.0 will deprecate all
+bundle management commands. The removal versions have not been set.
+
+Related issue: `#2113 <https://github.com/canonical/charmcraft/issues/2113>`_
+
+``bundle`` plugin
+~~~~~~~~~~~~~~~~~
+
+In line with the phasing out of bundles, Charmcraft 4.0 will deprecate the
+bundle plugin. No schedule has been set for the removal of the bundle plugin.
+
+Related issue: `#2114 <https://github.com/canonical/charmcraft/issues/2114>`_
+
+Known issues
+------------
+
+The following issues were reported and are scheduled to be fixed in upcoming
+patch releases.
+
+See individual issue links for any mitigations.
+
+- `#2078 <https://github.com/canonical/charmcraft/issues/2078>`_
+  ``charmcraft clean`` does not clean all platforms for a charm.
+- `#2012 <https://github.com/canonical/charmcraft/issues/2012>`_ Charmcraft uses
+  ``/cache`` as the cache directory if snapd doesn't set ``SNAP_USER_COMMON``
+- `#1990 <https://github.com/canonical/charmcraft/issues/1990>`_ Cannot stage
+  packages with Charmcraft
+
+
+Fixed bugs and issues
+---------------------
+
+The following issues have been resolved in Charmcraft 3.4:
+
+- `#2081 <https://github.com/canonical/charmcraft/issues/2081>`_
+  ``charmcraft pack`` fails because ``libffi-dev`` is missing.
+- `#2058 <https://github.com/canonical/charmcraft/issues/2058>`_ Multi-base charm
+  uses the same LXD container for different bases
+
+
+Contributors
+------------
+
+We would like to express a big thank you to all the people who contributed to
+this release.
+
+:literalref:`@bepri <https://github.com/bepri>`,
```

Comment:
@medubelko I'm not sure what to do about these linting issues:

```
  docs/release-notes/charmcraft-3.4.0.rst:143: missing underscore after closing backtick in hyperlink (missing-underscore-after-hyperlink)
  docs/release-notes/charmcraft-3.4.0.rst:144: missing underscore after closing backtick in hyperlink (missing-underscore-after-hyperlink)
  docs/release-notes/charmcraft-3.4.0.rst:145: missing underscore after closing backtick in hyperlink (missing-underscore-after-hyperlink)
```

Reply:
Resolved by telling sphinx-lint to be quiet about this syntax: https://github.com/canonical/charmcraft/pull/2115/commits/ac5f0171c4775c97daeb8c3a3cf17caf715e8a27

Taken from: https://github.com/canonical/rockcraft/blob/c8c5883ee8b98dd99f8d0efba8a443db4ea16336/common.mk#L165

I've upstreamed this fix: https://github.com/canonical/starbase/pull/333